Extensions/ArtifactEngine/Providers/webProvider.ts

```diff
@@ -39,26 +43,23 @@ export class WebProvider implements models.IArtifactProvider {
     }
 
     getArtifactItem(artifactItem: models.ArtifactItem): Promise<NodeJS.ReadableStream> {
-        var promise = new Promise<NodeJS.ReadableStream>(async (resolve, reject) => {
+        var promise = new Promise<NodeJS.ReadableStream>((resolve, reject) => {
             if (!artifactItem.metadata || !artifactItem.metadata['downloadUrl']) {
                 reject("No downloadUrl available to download the item.");
             }
 
             var itemUrl: string = artifactItem.metadata['downloadUrl'];
             itemUrl = itemUrl.replace(/([^:]\/)\/+/g, "$1");
-            var getResponsePromise = this.httpc.get(itemUrl);
-            getResponsePromise.catch(reason => {
+            this.httpc.get(itemUrl).then((res: httpm.HttpClientResponse) => {
+                if (res.message.headers['content-encoding'] === 'gzip') {
+                    resolve(res.message.pipe(zlib.createUnzip()));
```

Comment:
in case we don't catch exception from createUnzip, will it go to unhandled exception handler?

Reply:
yes